### PR TITLE
(partial) Transactions implementation

### PR DESCRIPTION
**Pull request by soerenbf**
## Purpose

Implements parts of the transactions possible to submit to a Concordium node

## Changes

- Added more transactions and the domain types used by them

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: Sources/Concordium/Domain/Block.swift [hunk 1]
```diff
@@ -1,15 +1,13 @@
 import Foundation
 
-public typealias BlockHash = Data // 32 bytes
-
-public enum BlockIdentifier {
+public enum BlockIdentifier: ToGRPC {
```
Comment (ramakser) [inline Sources/Concordium/Domain/Block.swift:3]: maybe we should update protocol naming. Make this protocol name like `GRPCConvertible`

File: Sources/Concordium/Domain/Block.swift [hunk 2]
```diff
@@ -1,15 +1,13 @@
...
     case lastFinal
     case best
     case hash(BlockHash)
     case absoluteHeight(UInt64)
     case relativeHeight(genesisIndex: UInt32, height: UInt64, restrictedToGenesisIndex: Bool)
 
-    func toGRPCType() -> Concordium_V2_BlockHashInput {
+    func toGRPC() -> Concordium_V2_BlockHashInput {
```
Comment (ramakser) [inline Sources/Concordium/Domain/Block.swift:10]: same here `Concordium_V2_BlockHashInput` its not swift naming convention. we use camelcase instead. 
`toGRPC` converts to `Concordium_V2_BlockHashInput` its not obvious why here gprc, maybe better to rename this function, to make to more self explanatory

**Review by ramakser — APPROVED** [review]

File: Sources/Concordium/Domain/Block.swift (hunk 1 above)
Comment (soerenbf) [reply Sources/Concordium/Domain/Block.swift:3]: Yeah I had something similar to that at one point, but then I realised not all types need to convert both ways (there is also a `FromGRPC` protocol.

In most cases we only need `FromGRPC` as that is meant for translating types returned from the GRPC API, whereas `ToGRPC` is meant for translating input into the types required by the generated GRPC client.

File: Sources/Concordium/Domain/Block.swift (hunk 2 above)
Comment (soerenbf) [reply Sources/Concordium/Domain/Block.swift:10]: These types (`Concordium_V2_*`) are from the code generated from the GRPC API. Also they're internal only as the generated code is not public, so no user will ever use this type when using the SDK.

Can we just keep them as is? 😄

File: Sources/Concordium/Domain/Block.swift (hunk 1 above)
Comment (ramakser) [reply Sources/Concordium/Domain/Block.swift:3]: Sure, let's leave it as it is for now. Perhaps later we can come up with a better name for this protocol.

**Pull request merged**